symengine/pow.h

```diff
@@ -34,12 +34,12 @@ class Pow : public Basic
     //! \return `true` if canonical
     bool is_canonical(const Basic &base, const Basic &exp) const;
     //! \return `base` of `base**exp`
-    inline RCP<const Basic> get_base() const
+    inline const RCP<const Basic>& get_base() const
     {
         return base_;
     }
     //! \return `exp` of `base**exp`
-    inline RCP<const Basic> get_exp() const
+    inline const RCP<const Basic>& get_exp() const
```

Comment:
The logic is that if you return a reference from a method, then the instance owns it, and you should not assign it anywhere, just use it, preferably on the same line, like this:
```c++
return get_exp().diff(x);
```
but not like this:
```c++
a = get_exp();
```

Now, you only return RCP, or accept RCP as an argument, if you want to manipulate the reference counter (refcount), in other words, if you want to assign it to another RCP variable. In all other cases, you accept or return the object itself (as a reference).

So it depends on how get_exp() is used in symengine. If it is used to assign the result somewhere else, then it must be defined as:
```c++
RCP<const Basic> get_exp()
``
or it is used only to look at the result, but not assign it, in which case it should be declared as:
```c++
const Basic& get_exp()
```

The `const RCP<const Basic>&` doesn't make any sense to me. It is also against the Trilinos Teuchos recommendations.